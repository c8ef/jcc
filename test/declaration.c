int main() {
  int x;
  int y = 42;

  int a, b, c;
  int m, n, o = 42;
}
