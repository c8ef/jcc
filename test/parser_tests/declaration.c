int main() {
  int x;
  int y = 42;

  int a, b, c;
  int m, n, o = 42;

  int q;
  q = 42;
}
