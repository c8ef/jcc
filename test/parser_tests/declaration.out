FunctionDecl: main
  Args[0]
  CompoundStatement
    DeclStatement
      VarDecl: x
    DeclStatement
      VarDecl: y
        IntergerLiteral: 42
    DeclStatement
      VarDecl: a
      VarDecl: b
      VarDecl: c
    DeclStatement
      VarDecl: m
        IntergerLiteral: 42
      VarDecl: n
        IntergerLiteral: 42
      VarDecl: o
        IntergerLiteral: 42
    DeclStatement
      VarDecl: q
    ExprStatement
      BinaryExpr(=):
        DeclRefExpr: q
        IntergerLiteral: 42
